pybamm/models/submodels/interface/sei/base_sei.py

```diff
@@ -230,7 +230,10 @@ def _get_standard_concentration_variables(self, variables):
         elif self.reaction == "SEI on cracks":
             L_inner_cr = variables["Inner SEI on cracks thickness"]
             L_outer_cr = variables["Outer SEI on cracks thickness"]
-            roughness = variables[self.domain + " electrode roughness ratio"]
+            if self.domain + " electrode roughness ratio" in variables:
```

Comment:
if the sei submodel is called before the roughness submodel, then the roughness would be set to 1 even though the roughness variable would be later added

Reply:
I don't understand why no_mechanics isn't setting the roughness to 1 anyway

Reply:
I set two breakpoints in no_mechanics. One was on line 30 in get_fundamental_variables, and that got triggered. The other was in line 40 in get_coupled_variables, which did not get triggered. Could it be to do with the order in which the models are built?

Reply:
The submodels get built in a loop so the order shouldn't matter. Usually when something like this happens it is because submodel A requires a variable from submodel B and submodel B requires a variable from submodel A. You could try running with `pybamm.set_logging_level("DEBUG")` and it will print which variables it can't find.

Reply:
Last two before failing:
2022-06-23 16:20:34.510 - [DEBUG] base_battery_model.build_fundamental_and_external(760): Getting fundamental variables for sei submodel (Doyle-Fuller-Newman model)
2022-06-23 16:20:34.510 - [DEBUG] base_battery_model.build_fundamental_and_external(760): Getting fundamental variables for sei on cracks submodel (Doyle-Fuller-Newman model)